src/librustc/traits/specialize/specialization_graph.rs

```diff
@@ -345,14 +345,27 @@ impl<T> NodeItem<T> {
 impl<'a, 'gcx, 'tcx> Ancestors {
     /// Search the items from the given ancestors, returning each definition
     /// with the given name and the given kind.
+    // FIXME: Migrate all uses of this function to `defs_hygienic` (or equivalent)
+    // to support macros 2.0.
     #[inline] // FIXME(#35870) Avoid closures being unexported due to impl Trait.
-    pub fn defs(self, tcx: TyCtxt<'a, 'gcx, 'tcx>, name: Name, kind: ty::AssociatedKind)
+    pub fn defs_unhygienic(self, tcx: TyCtxt<'a, 'gcx, 'tcx>, name: Name, kind: ty::AssociatedKind)
                 -> impl Iterator<Item = NodeItem<ty::AssociatedItem>> + 'a {
         self.flat_map(move |node| {
             node.items(tcx).filter(move |item| item.kind == kind && item.name == name)
                            .map(move |item| NodeItem { node: node, item: item })
         })
     }
+    #[inline] // FIXME(#35870) Avoid closures being unexported due to impl Trait.
+    pub fn defs_hygienic(self, tcx: TyCtxt<'a, 'gcx, 'tcx>, name: Name,
+                         kind: ty::AssociatedKind, trait_def_id: DefId, impl_node_id: NodeId)
+                         -> impl Iterator<Item = NodeItem<ty::AssociatedItem>> + 'a {
+        self.flat_map(move |node| {
+            node.items(tcx).filter(move |item| {
+                let item_ident = tcx.adjust(item.name, trait_def_id, impl_node_id).0;
```

Comment:
Hm, actually `adjust` doesn't require the `NodeId` if we need only the adjusted ident, so it looks like specialization can be fixed as well.